File: output/xCodeEval_8000/C++/Go/4fa18e57c15335beb869eade8efcb49d.go
package main

import (
	"fmt"
	"math"
)

type int_uv struct {
	u int
	v int
}

type int_abc struct {
	a int
	b int
	c int
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func reverse(path []int) {
	n := len(path)
	for i := 0; i < n/2; i++ {
		path[i], path[n-i-1] = path[n-i-1], path[i]
	}
}

func pathFrom(a, b []int, v int, f func(int) bool, path []int, markers []int) bool {
	if f(v) {
		path = append(path, v)
		return true
	}
	markers[v] = 1
	res := false
	for _, e := range path[v] {
		if markers[e.v] == 0 && pathFrom(a, b, e.v, f, path, markers) {
			path = append(path, v)
			res = true
			break
		}
	}
	return res
}

func solve() {
	var n, vol, e int
	fmt.Scan(&n, &vol, &e)

	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
	}

	b := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&b[i])
	}

	ee := make([]int_uv, e)
	for i := 0; i < e; i++ {
		fmt.Scan(&ee[i].u, &ee[i].v)
	}

	G := make([][]int_uv, n)
	for _, edge := range ee {
		u, v := edge.u-1, edge.v-1
		G[u] = append(G[u], int_uv{u + 1, v + 1})
		G[v] = append(G[v], int_uv{v + 1, u + 1})
	}

	G2 := make([][]int_uv, n)
	markers := make([]int, n)
	for i := 0; i < n; i++ {
		G2[i] = make([]int_uv, len(G[i]))
		copy(G2[i], G[i])
	}

	ee2 := make([]int_uv, 0)
	for i := 0; i < n; i++ {
		if a[i] != b[i] {
			path := make([]int, 0)
			if !pathFrom(a, b, i, func(v int) bool { return (a[v]-b[v])%(a[i]-b[i]) < 0 }, path, markers) {
				fmt.Println("NO")
				return
			}
			if a[path[0]] < b[path[0]] {
				reverse(path)
			}

			src := path[0]
			tgt := path[len(path)-1]
			diff := min(abs(a[src]-b[src]), abs(a[tgt]-b[tgt]))
			doit(a, b, path, res, 0, vol, diff)
			a[src] -= diff
			a[tgt] += diff
		}
	}

	fmt.Print(len(res))
	for _, x := range res {
		fmt.Printf("%d %d %d", x.a, x.b, x.c)
	}
}

func main() {
	solve()
}

File: output/xCodeEval_remove_8000/C++/C/4fa18e57c15335beb869eade8efcb49d.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int a;
    int b;
    int c;
} int_abc;

typedef struct {
    int u;
    int v;
} int_uv;

void doit(const int *a, const int *b, const int *path, int_abc *res, int pos, int vol, int diff, int path_len) {
    if (pos + 1 >= path_len) return;
    int tgt = path[pos + 1];
    int before = (diff < vol - a[tgt]) ? diff : vol - a[tgt];
    int src = path[pos];
    if (before > 0) {
        res->a = src + 1;
        res->b = tgt + 1;
        res->c = before;
        res++;
    }
    doit(a, b, path, res, pos + 1, vol, diff, path_len);
    int after = diff - before;
    if (after > 0) {
        res->a = src + 1;
        res->b = tgt + 1;
        res->c = after;
        res++;
    }
}

void solve() {
    int n, vol, e;
    scanf("%d %d %d", &n, &vol, &e);

    int *a = (int*)malloc(n * sizeof(int));
    int *b = (int*)malloc(n * sizeof(int));
    int_uv *ee = (int_uv*)malloc(e * sizeof(int_uv));

    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }

    for (int i = 0; i < n; i++) {
        scanf("%d", &b[i]);
    }

    for (int i = 0; i < e; i++) {
        scanf("%d %d", &ee[i].u, &ee[i].v);
    }

    int_uv *ee2 = (int_uv*)malloc(e * sizeof(int_uv));
    int *markers = (int*)malloc(n * sizeof(int));

    for (int i = 0; i < n; i++) {
        ee2[i].u = ee[i].u + 1;
        ee2[i].v = ee[i].v + 1;
        markers[i] = 0;
    }

    int *path = (int*)malloc(n * sizeof(int));
    int path_len = 0;

    for (int q = 0; q < n; q++) {
        for (int i = 0; i < n; i++) {
            if (a[i] != b[i]) {
                path_len = 0;
                markers[i] = 1;
                path[path_len++] = i;
                int v = i;
                while (1) {
                    int found = 0;
                    for (int j = 0; j < e; j++) {
                        if (ee2[j].u == v && (a[ee2[j].v - 1] - b[ee2[j].v - 1]) % (a[i] - b[i]) < 0) {
                            v = ee2[j].v - 1;
                            markers[v] = 1;
                            path[path_len++] = v;
                            found = 1;
                            break;
                        }
                    }
                    if (!found) {
                        break;
                    }
                }
                if (a[path[0]] < b[path[0]]) {
                    for (int j = 0; j < path_len / 2; j++) {
                        int temp = path[j];
                        path[j] = path[path_len - j - 1];
                        path[path_len - j - 1] = temp;
                    }
                }
                int src = path[0];
                int tgt = path[path_len - 1];
                int diff = (a[src] - b[src] < a[tgt] - b[tgt]) ? a[src] - b[src] : a[tgt] - b[tgt];
                int_abc *res = (int_abc*)malloc(n * sizeof(int_abc));
                doit(a, b, path, res, 0, vol, diff, path_len);
                a[src] -= diff;
                a[tgt] += diff;
                free(res);
            }
        }
    }

    printf("%d\n", n);
    for (int i = 0; i < n; i++) {
        printf("%d %d %d\n", res[i].a, res[i].b, res[i].c);
    }

    free(a);
    free(b);
    free(ee);
    free(ee2);
    free(markers);
    free(path);
}

int main() {
    solve();
    return 0;
}
